core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -95,10 +98,17 @@
      */
     @BeforeClass
     public static void init() throws IOException {
+        Version version = Version.CURRENT;
+        // sometimes we need pre-1.3 index version to simulate disabled FieldNamesFieldMapper (e.g. for ExistQueryBuilder)
+        if (randomIntBetween(0, 5) == 0) {
+            version = Version.V_1_2_0;
+        }
```

Comment:
shall we simply randomize the version at this point between a lower bound and current? this one replaces what you were doing before by manually disabling or enabling the mapper right?